ibm/resource_ibm_cis_certificate_upload.go

```diff
@@ -0,0 +1,354 @@
+		certsList := []cissslv1.CertPriorityReqCertificatesItem{}
+		id := certID
+		var priority int64
+		if v, ok := d.GetOk(cisCertificateUploadPriority); ok {
+			priority = int64(v.(int))
+		}
+		certsItem, _ := cisClient.NewCertPriorityReqCertificatesItem(id, priority)
+		certsList = append(certsList, *certsItem)
+		priorityOpt := cisClient.NewChangeCertificatePriorityOptions()
+		priorityOpt.SetCertificates(certsList)
+		priorityResponse, err := cisClient.ChangeCertificatePriority(priorityOpt)
+		if err != nil {
+			log.Printf("Change certificate priority failed: %v", priorityResponse)
+			return err
+		}
+	}
+	return resourceCISCertificateUploadRead(d, meta)
+}
+
+func resourceCISCertificateUploadDelete(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+
+	certID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+	opt := cisClient.NewDeleteCustomCertificateOptions(certID)
+	response, err := cisClient.DeleteCustomCertificate(opt)
+	if err != nil {
+		log.Printf("Delete custom certificate failed: %v", response)
+		return err
+	}
+	_, err = waitForCISCertificateUploadDelete(d, meta)
+	if err != nil {
+		return err
+	}
+	d.SetId("")
+	return nil
+}
+
+func resourceCISCertificateUploadExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return false, err
+	}
+
+	certID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return false, err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	opt := cisClient.NewGetCustomCertificateOptions(certID)
+	_, response, err := cisClient.GetCustomCertificate(opt)
+	if err != nil {
+		if response != nil && response.StatusCode == 400 {
+			log.Printf("custom certificate does not exist.")
+			return false, nil
+		}
+		log.Printf("Get custom certificate failed: %v", response)
+		return false, err
+	}
+	return true, nil
+}
+
+func waitForCISCertificateUploadDelete(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return nil, err
+	}
+
+	certID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return nil, err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	opt := cisClient.NewGetCustomCertificateOptions(certID)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{cisCertificateUploadDeletePending},
+		Target:  []string{cisCertificateUploadDeleted},
+		Refresh: func() (interface{}, string, error) {
+			_, detail, err := cisClient.GetCustomCertificate(opt)
+			if err != nil {
+				if detail != nil && detail.StatusCode == 400 {
```

Comment:
should be 404

Reply:
Get request does not return 404 error code. it only return 400 as invalid certificate error. So I removed 400 check from Read() method. Kept the check for wait logic.